### ISSUE TITLE
Fix and improve gas estimation in new-ui send screen 

### ISSUE THREAD

**Issue by danjm** [MetaMask/metamask-extension#3914]
We need to fix and then improve gas estimation in the new-ui send screen.

Requirements include:

- [ ] use the same algorithm for calculating gas as is used in old-ui, be sure to pass same data (when available) to `estimateGas`
- [ ] on component mount, we need to make an initial estimate without a to address
- [ ] when the to address changes, we need to update the estimate
- [ ] prevent the user from customizing the limit to a value below the minimum required given the estimate
- [ ] if the user customizes the gas limit and then changes the to address, only change the gas limit if it is below the minimum required according to the new estimate
- [ ] Handle the special case of estimating for token transfers.
    - [ ] correct value in the `to` address passed to estimateGas
    - [ ] (?) update 'data' when 'To' or amount fields in send form changes (?)
- [ ] Nice to have: update the estimate if blockGasLimit updates
- [ ] Nice to have: give user an indication that the gas limit has been updated

**Comment by danjm** [comment]
Blocked by #3725
